lib/inc/hocon/config_mergeable.hpp

```diff
@@ -6,6 +6,7 @@
 namespace hocon {
 
     class LIBCPP_HOCON_EXPORT config_mergeable {
+        friend class config_value;
```

Comment:
Because of one place in config_value where we do:

```
auto other = dynamic_pointer_cast<const config_mergeable>(mergeable)->to_fallback_value();
```

and if config_value isn't a friend class, we get:

```
/Users/whopper/Coding/cpp-hocon/lib/src/values/config_value.cc:124:83: error: 'to_fallback_value' is a protected member of 'hocon::config_mergeable'
            auto other = dynamic_pointer_cast<const config_mergeable>(mergeable)->to_fallback_value();
                                                                                  ^
/Users/whopper/Coding/cpp-hocon/lib/inc/hocon/config_mergeable.hpp:68:30: note: can only access this member on an object of type 'hocon::config_value'
        virtual shared_value to_fallback_value() const = 0;
```